src/Explorer/Tabs/Tabs.tsx

```diff
@@ -158,6 +183,7 @@ const CloseButton = ({
     onClick={(event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
       event.stopPropagation();
       tab ? tab.onCloseTabButtonClick() : useTabs.getState().closeReactTab(tabKind);
+      tabKind === ReactTabKind.QueryCopilot && queryCopilotState.resetQueryCopilotStates();
```

Comment:
As mentioned above, simply call `useQueryCopilotState.getState().resetQueryCopilotStates()` here

Reply:
getState() does not exist because useQueryCopilotState is a custom hook with states and functions. I will try to fix it in another way.